### PR TITLE
Upgrading utf8 comparisons and sha256 tests

### PR DESCRIPTION
**Pull request by pschuh**
test82.in -> test_utf8_comparisons.vim
test90.in -> test_sha256.vim

### CODE REVIEW

**Comment by chrisbra** [comment]
oh wow, that are a lot of changes. Can you please rename the new numeric tests like test24.vim to a better name, e.g. test23.vim could be named test_ex_edit.vim, test24.vim -> test_regex_backslash.vim, test26.vim -> test_exe_while_if.vim, test67.vim -> test_exists.vim, test75.vim -> test_maparg.vim, test82.vim -> test_mbyte_strnicmp.vim, test90.vim -> test_sha256vim, test97.vim -> test_glob.vim. 
My suggestions might not be the best, however the names should be descriptive so use whatever you think fits best. And it should be sorted in the Makefile. Thanks.

File: src/testdir/test_getcwd.vim [hunk 1]
```diff
@@ -0,0 +1,91 @@
+function! GetCwdInfo(win, tab)
+ 	let tab_changed = 0
+ 	let mod = ":t"
+ 	if a:tab > 0 && a:tab != tabpagenr()
+   	let tab_changed = 1
+   	exec "tabnext " . a:tab
```
Comment (dpelle) [inline src/testdir/test_getcwd.vim]: It's a bit hard to read without indenting the if and else blocks.

File: src/testdir/test_getcwd.vim [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   	let dirname = fnamemodify(getcwd(a:win), mod)
+   	let lflag = haslocaldir(a:win)
+ 	else
+   	let dirname = fnamemodify(getcwd(a:win, a:tab), mod)
+   	let lflag = haslocaldir(a:win, a:tab)
+ 	endif
+ 	return bufname . ' ' . dirname . ' ' . lflag
+endfunction
+
+" Do all test in a separate window to avoid E211 when we recursively
```
Comment (dpelle) [inline src/testdir/test_getcwd.vim]: all test → all tests

**Review by dpelle — COMMENTED** [review]
Thanks!
Consider creating several independent
pull requests rather than a big one.

**Comment by brammool** [comment]
pschuh wrote:

> # case insensitive utf8 comparisons 
> test82.vim
> # sha256 
> test90.vim 
> # escaped glob() 
> test97.vim 
> test_getcwd.vim 
> # + arg to edit 
> test23.vim 
> # regexps with special [] chars 
> test24.vim 
> # test execute while if 
> test26.vim 
> # exists() autocommand 
> test67.vim 
> # maparg() 
> test75.vim 
> test_lua.vim 
> test_comparators.vim

Thanks for making the tests better!
Doing this in smaller pieces is indeed easier for me.

When we convert the test from old style to new style, we should also try
to make them as readable as we can. Please change the literal Escape
into "\e":

call feedkeys("iabcd\n\ndefggg0g~ap", "tx")

To:

call feedkeys("iabcd\n\ndefg\egg0g~ap", "tx")

I'll make that change now, please also keep an eye out for this in other
tests.

-- 
From "know your smileys":
 ...---...   SOS

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Pull request closed without merging**